### PR TITLE
feat: consent manager us opt in

### PR DESCRIPTION
**Pull request by alexcarpenter**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-acconsent-manager-us-opt-in-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/home/1200641486636491/1202671127598803) 🎟️

## 🗒️ What

Implements latest consent-manger updates which includes auto opt-in for folks within the US. 

consent-manager PR: https://github.com/hashicorp/react-components/pull/658

## 🛠️ How

Bumps

## 📸 Design Screenshots

CleanShot 2022-08-01 at 14 39 03@2x

## 💭 Anything else?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202671127598803
  - https://app.asana.com/0/0/1202671127598804

### CODE REVIEW

File: src/pages/_middleware.ts [hunk 1]
```diff
@@ -70,7 +73,7 @@ export function middleware(req: NextRequest, ev: NextFetchEvent) {
 	// Handle Opt-in cookies
 	const optInPlatform = params.get('optInFrom') as OptInPlatformOption
 	const hasOptedIn = Boolean(req.cookies[`${optInPlatform}-beta-opt-in`])
-	const response = NextResponse.next()
+	response = NextResponse.next()
```
Comment (alexcarpenter) [inline src/pages/_middleware.ts:76]: This doesn't seem to work as I was expecting. @BRKalow Curious how to use `setGeoCookie` with `NextResponse.next()` usage. Will need to use the same setup on Learn https://github.com/hashicorp/learn/pull/4504/files#diff-db97c6a1b4af3a37ea8a8a1786dc326bb1e26cdd70b92225480081e61be5b552R83

Comment (BRKalow) [reply src/pages/_middleware.ts:76]: This line creates a new response, in order for the cookie util to work we need to re-use the response that's declared from `setGeoCookie(req)`. We should be able to remove this line.

Comment (alexcarpenter) [reply src/pages/_middleware.ts:76]: Got it, so `NextResponse.next()` is not needed?

Comment (BRKalow) [reply src/pages/_middleware.ts:76]: Correct 👍

**Pull request merged**